xdsl/ir/affine/affine_expr.py

```diff
@@ -159,9 +159,42 @@ def eval(self, dims: list[int], symbols: list[int]) -> int:
 
         raise ValueError("Unreachable")
 
+    def _try_fold_constant(
+        self, other: AffineExpr, binop: _AffineExprKind
+    ) -> AffineExpr | None:
+        if self._impl.kind != _AffineExprKind.Constant:
+            return None
+        if other._impl.kind != _AffineExprKind.Constant:
+            return None
+        assert isinstance(self._impl, _AffineConstantExprStorage)
+        assert isinstance(other._impl, _AffineConstantExprStorage)
+
+        match binop:
+            case _AffineExprKind.Add:
+                return AffineExpr.constant(self._impl.value + other._impl.value)
+            case _AffineExprKind.Mul:
+                return AffineExpr.constant(self._impl.value * other._impl.value)
+            case _AffineExprKind.Mod:
+                return AffineExpr.constant(self._impl.value % other._impl.value)
+            case _AffineExprKind.FloorDiv:
+                return AffineExpr.constant(self._impl.value // other._impl.value)
+            case _AffineExprKind.CeilDiv:
+                return AffineExpr.constant(-(-self._impl.value // other._impl.value))
+            case _:
+                raise ValueError(f"Unreachable")
+
     def __add__(self, other: AffineExpr | int) -> AffineExpr:
         if isinstance(other, int):
             other = AffineExpr.constant(other)
+
+        # Fold constants.
+        if fold := self._try_fold_constant(other, _AffineExprKind.Add):
+            return fold
+
+        # Keep the constant on the right side.
+        if self._impl.kind == _AffineExprKind.Constant:
+            self, other = other, self
```

Comment:
That's so cursed ;) (But that's fine)

Reply:
Any better way to swap things?

Reply:
No that's the best!
The only curse thing in my opinion is swapping `self`, but that's fine!

Reply:
Maybe something like `other.__add__(self)`?